### PR TITLE
Celery updates

### LINKED ISSUE #101: Update tests to support updated celery strategy
The updated celery transformation strategy in `oteapi-core` is wreaking havoc on the tests here due to attempt connections on REDIS. This needs to be mitigated, either by mocking or otherwise, depending on the individual tests.

### PR DESCRIPTION
**Pull request by CasperWA**
# Description:

Fixes #101 

The tests have gotten an overhaul, modularizing the pytest fixtures further, while collecting all strategy tests (since they were extremely similar) in a single file, supporting running through all backends as well.

The general backend split has been tightened as well, moving parts "upstream" (e.g., from a specific strategy to the base strategy class or from the base strategy class to the backend client class, etc.)

Implement Celery-specific support for testing the `celery/remote` strategy and the updates it has received in OTEAPI Core.

## Type of change:

- [x] Bug fix.
- [ ] New feature.
- [ ] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: tests/strategies/test_generic_tests.py
Comment (daniel-sintef) [inline tests/strategies/test_generic_tests.py:1]: The name seems a bit weird, maybe should be test_generic_functions?

**Comment by daniel-sintef** [comment]
"Looks Good"^TM. Unfortunately its just a lot of code for me to digest right now. My schedule is pretty packed the next few days but I could book an hour to look at it in more detail on Wednesday. I am otherwise OK to merge it now, but its just too much for me to think deeply about at the moment.

File: tests/strategies/test_generic_tests.py
Comment (CasperWA) [reply tests/strategies/test_generic_tests.py:1]: I suppose. The intended name is "Generic tests", but pytest requires all test files to start with `test_`.
I guess it could be `test_all_strategies` instead?

Comment (daniel-sintef) [reply tests/strategies/test_generic_tests.py:1]: yes that would be much better!

**Comment by CasperWA** [comment]
> "Looks Good"^TM. Unfortunately its just a lot of code for me to digest right now. My schedule is pretty packed the next few days but I could book an hour to look at it in more detail on Wednesday. I am otherwise OK to merge it now, but its just too much for me to think deeply about at the moment.

I can't merge until someone approves it.

**Comment by daniel-sintef** [comment]
@CasperWA am I able to approve? In the 'review required' section it only shows @quaat as a reviewer

**Comment by CasperWA** [comment]
> @CasperWA am I able to approve? In the 'review required' section it only shows @quaat as a reviewer

It just shows him in the list of "1 pending reviewer", because he hasn't reviewed it yet. But you have. You should be able to approve by doing another review normally.

**Review by daniel-sintef — APPROVED** [review]

**Pull request merged**
